tests/e2e/autoscale/04-delete.yaml

```diff
@@ -0,0 +1,7 @@
+apiVersion: kuttl.dev/v1beta1
```

Comment:
Why is this needed? The https://github.com/open-telemetry/opentelemetry-operator/blob/0232b80b1885bf616d4cc5134c6422ed51f7199b/tests/e2e/autoscale/01-install.yaml#L4 should be cleaned automatically.

Reply:
The problem is: depending on the cluster where you are running the test, the previous `duration` parameter provided to `tracegen` can be not enough to trigger the HPA and scale the collector.

With this approach, we ran `tracegen` until the number of replicas is increased. Later, we stop `tracegen` to reduce the metrics and make the HPA to scale down.